flink-core/src/main/java/org/apache/flink/util/SerializedValue.java

```diff
@@ -63,6 +66,7 @@ public T deserializeValue(ClassLoader loader) throws IOException, ClassNotFoundE
 	 *
 	 * @return Serialized data.
 	 */
+	@Nullable
```

Comment:
The BlobWriter and SerializedValueSerializer do not handle the nullable case appropriately. It seems as if no user of this class so far was properly handling this case, which makes me wonder whether it a) should never return a null array or b) not allow null values in the first place.

Reply:
Yes, there are indeed other problems with this class. I would consider this follow up tasks.

Reply:
Does it make sense to create a Jira issue to cover the null handling in the classes related to this change?

Reply:
Yes, I think so.

Reply:
I created it: [FLINK-20580](https://issues.apache.org/jira/browse/FLINK-20580)

Reply:
Thanks @XComp.